polyphemus/lib/etls/add_watch_folder_base_etl.rb

```diff
@@ -1,48 +1,86 @@
 require_relative "./metis_folder_filtering_base_etl"
+require_relative './watch_etl_config'
 
 class Polyphemus::AddWatchFolderBaseEtl < Polyphemus::MetisFolderFilteringBaseEtl
-  def initialize(project_bucket_pairs:, model_name:, folder_path_regexes: {}, limit: 20, watch_type:, linker:)
-    @linker = linker
-    @model_name = model_name
-    @watch_type = watch_type
+  def initialize(bucket_watch_configs: [], limit: 20)
+    pairs = bucket_watch_configs.map(&:cursor_pair)
+    raise "bucket_watch_configs have non unique project/bucket combinations" unless pairs.uniq.length == pairs.length
+
+    @bucket_watch_configs = bucket_watch_configs
     super(
-      project_bucket_pairs: project_bucket_pairs,
+      project_bucket_pairs: pairs,
       limit: limit,
-      folder_path_regexes: folder_path_regexes,
+      folder_path_regexes: bucket_watch_configs.map do |config|
+        [
+          project_bucket_symbol(
+            project_name: config.project_name,
+            bucket_name: config.bucket_name,
+          ),
+          config.sum_bucket_regex,
+        ]
+      end.to_h
     )
   end
 
+  def bucket_config(cursor)
+    @bucket_watch_configs.find { |b| b.bucket_name == cursor[:bucket_name] && b.project_name == cursor[:project_name] }.tap do |config|
+      if config.nil?
+        raise "Could not find bucket_config matching cursor, either reset the cursor or fix the config"
+      end
+    end
+  end
+
+  def watch_type_groups(cursor, folders)
+    return {} if bucket_config(cursor).nil?
+
+    {}.tap do |watch_type_groups|
+      folders.each do |folder|
+        bucket_config(cursor).find_matching_watches(folder.folder_path).each do |watch_config|
+          folders = watch_type_groups[watch_config.watch_type] ||= []
+          folders << folder
+        end
+      end
+    end
+  end
+
   def process(cursor, folders)
     # We'll need to filter out the folders based on folder_name here, using the
     #   supplied regexes
-    logger.info("Found #{folders.length} updated folders: #{folders.map { |f| f.folder_path }.join(",")}")
+    logger.info("Found #{folders.length} updated matching folders: #{folders.map { |f| f.folder_path }.join(",")}")
```

Comment:
oh you know what, I can revert this.  I had previously tried making filtering a step up from process function but changed gears, so this is an arifact.